crates/astria-conductor/src/conductor.rs

```diff
@@ -245,6 +269,16 @@ impl Conductor {
         }
         info!("shutting down");
     }
+
+    #[instrument(skip_all, err)]
+    async fn restart(&mut self) -> eyre::Result<()> {
```

Comment:
Couple notes on this logic:

1. if the `run_until_stopped` loop introduces logic before it enters its event loop, then this restart function will immediately break.
2. this is hard killing the other tasks without waiting for them to complete. The line `self.tasks = new_conductor.tasks` just aborts all the functions immediately. It's not unlikely that `self.task_shutdown.cancel()` was not yet received by the other tasks.
3. this code should allow the other tasks to exit gracefully before starting a new conductor.

Reply:
Added wrapper around Conductor per your suggestions, allowing the inner conductor task to fully shut down and return `RestartToken::Restart` if it should be restarted, or `RestartToken::Shutdown` if not. Per d6fde21. Thanks for your help on this!